tools/marvin/marvin/lib/base.py

```diff
@@ -1521,12 +1513,12 @@ def _set_command(cls, apiclient, cmd, services, random_name=True):
             raise Exception(
                 "Unable to find Ostype is required for creating template")
 
-        cmd.snapshotid = snapshot.id
+        if volume:
+            cmd.volumeid = volume.id
+
+        if snapshot:
+            cmd.snapshotid = snapshot.id
 
-        if account:
```

Comment:
@slavkap dont' we need them anymore?

Reply:
@rhtyd, yes, there is no need to provide account/domain, according to the  - [API docs ](https://cloudstack.apache.org/api/apidocs-4.15/apis/createTemplate.html)
also, I've checked the code